src/context/cdlist.h

```diff
@@ -384,7 +384,7 @@ class CDList : public ContextObj {
 
     // FIXME we support operator--, but do we satisfy all the
     // requirements of a bidirectional iterator ?
-    typedef std::input_iterator_tag iterator_category;
+    typedef std::forward_iterator_tag iterator_category;
```

Comment:
This is actually bidirectional.
```suggestion
    typedef std::bidirectional_iterator_tag iterator_category;
```